### ISSUE TITLE
How to return logits when do_sample=True, temperature >0

### ISSUE THREAD

**Issue by ppalantir** [QwenLM/Qwen3-VL#1777]
outputs_gen = model.generate(
                    **inputs,
                    max_new_tokens=4096,
                    temperature=0.2,
                    do_sample=False,
                    output_scores=True,  
                    return_dict_in_generate=True 
                ) 

many elements in outputs_gen.scores are -inf

**Comment by JJJYmmm** [comment]
could you provide more detailed information, such as model size? A minimal test script would be preferred.
